tests/gradients/core/test_pulse_gradient.py

```diff
@@ -264,6 +265,278 @@ def test_with_parametrized_evolution(self):
             assert qml.equal(t.operations[-1], ops[2])
 
 
+@pytest.mark.jax
+class TestParshiftAndIntegrate:
+    """Test the helper routine ``_parshift_and_integrate``. Most tests use uniform
+    return types and parameters, so that we can test against simple tensor contractions."""
+
+    # pylint: disable=too-many-arguments
+
+    @pytest.mark.parametrize("multi_term", [1, 4])
+    @pytest.mark.parametrize("meas_shape", [(), (4,)])
+    @pytest.mark.parametrize("par_shape", [(), (3,), (2, 7)])
+    @pytest.mark.parametrize("num_shifts", [2, 5])
+    @pytest.mark.parametrize("num_split_times", [1, 7])
+    @pytest.mark.parametrize("use_broadcasting", [False, True])
```

Comment:
I guess this comes down to personal preference, so dont take my words for gospel and feel free to ignore it. I find highly parametrized tests very difficult to follow and would suggest, as a rule of thumb, to only use parametrizations that dont necessitate any related `if` clauses in the test.

Reply:
In general I fully agree with this. The reason I went for it nonetheless is the sheer amount of parameters. 
I'll see whether pulling out `multi_term` or `use_broadcasting` helps :)

Reply:
Pulled out `use_broadcasting`.